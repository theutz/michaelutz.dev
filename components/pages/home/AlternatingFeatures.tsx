/* eslint-disable @next/next/no-img-element */
import Image from "next/image";
import { ThumbUpIcon, GlobeIcon } from "@heroicons/react/outline";
import {
  useEndorsement,
  Endorsement as EndorsementType,
} from "hooks/useEndorsement";
import { useState } from "react";
import { classNames } from "utils/classNames";

function Endorsement({
  name,
  title,
  company,
  content,
  image,
}: EndorsementType) {
  const [showFull, setShowFull] = useState(false);
  return (
    <blockquote>
      <div onClick={() => setShowFull((prev) => !prev)}>
        <p
          className={classNames(
            "text-base text-gray-500 block whitespace-pre-line cursor-pointer",
            !showFull && "line-clamp-3"
          )}
        >
          &ldquo;{content.trim()}&rdquo;
        </p>
      </div>
      <footer className="mt-3">
        <div className="flex items-center space-x-3">
          <div className="relative flex-shrink-0">
            <div className="w-6 h-6">
              <Image
                className="rounded-full"
                layout="fill"
                objectFit="contain"
                src={image}
                alt={name}
              />
            </div>
          </div>
          <div className="text-base font-medium text-gray-700">
            {name}, {title} @ {company}
          </div>
        </div>
      </footer>
    </blockquote>
  );
}

export function AlternatingFeatureSections() {
  const { endorsement, increment } = useEndorsement();
  return (
    <div className="relative pt-16 pb-32 overflow-hidden">
      <div
        aria-hidden="true"
        className="absolute inset-x-0 top-0 h-48 bg-gradient-to-b from-gray-100"
      />
      <div className="relative">
        <div className="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24">
          <div className="max-w-xl px-4 mx-auto sm:px-6 lg:py-16 lg:max-w-none lg:mx-0 lg:px-0">
            <div>
              <div>
                <span className="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600">
                  <ThumbUpIcon
                    className="w-6 h-6 text-white"
                    aria-hidden="true"
                  />
                </span>
              </div>
              <div className="mt-6">
                <h2 className="text-3xl font-extrabold tracking-tight text-gray-900">
                  The Humans Have Spoken
                </h2>
                <p className="mt-4 text-lg text-gray-500">
                  Building tech products is also about building relationships.
                  Here&apos;s what some of my co-workers have to say about
                  working with me.
                </p>
                <div className="mt-6">
                  <button
                    onClick={() => increment()}
                    className="flex inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                  >
                    Read Another
                  </button>
                </div>
              </div>
            </div>
            <div className="pt-6 mt-8 border-t border-gray-200">
              <Endorsement {...endorsement} />
            </div>
          </div>
          <div className="mt-12 sm:mt-16 lg:mt-0">
            <div className="pl-4 -mr-48 sm:pl-6 md:-mr-16 lg:px-0 lg:m-0 lg:relative lg:h-full">
              <div className="relative lg:m-0 lg:h-full">
                <img
                  className="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:left-0 lg:h-full lg:w-auto lg:max-w-none filter brightness-90"
                  src="/assets/images/seattle-skyline.jpg"
                  alt="Inbox user interface"
                />
                <div className="absolute inset-0 mix-blend-overlay bg-gradient-to-r from-purple-800 to-indigo-600 lg:w-massive"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="mt-24">
        <div className="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24">
          <div className="max-w-xl px-4 mx-auto sm:px-6 lg:py-32 lg:max-w-none lg:mx-0 lg:px-0 lg:col-start-2">
            <div>
              <div>
                <span className="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600">
                  <GlobeIcon
                    className="w-6 h-6 text-white"
                    aria-hidden="true"
                  />
                </span>
              </div>
              <div className="mt-6">
                <h2 className="text-3xl font-extrabold tracking-tight text-gray-900">
                  Better understand your developer
                </h2>
                <p className="mt-4 text-lg text-gray-500">
                  I was born in lovely Seattle, Washington in the USA. I met my
                  wife in Nashville, Tennessee, and we&apos;ve moved to
                  Helsinki, Finland to start a family.
                </p>
                <div className="mt-6">
                  <a
                    href="#"
                    className="inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                  >
                    Get started
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div className="mt-12 sm:mt-16 lg:mt-0 lg:col-start-1">
            <div className="pr-4 -ml-48 sm:pr-6 md:-ml-16 lg:px-0 lg:m-0 lg:relative lg:h-full">
              <div className="relative lg:h-full lg:m-0">
                <img
                  className="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:right-0 lg:h-full lg:w-auto lg:max-w-none"
                  src="/assets/images/helsinki-skyline.jpg"
                  alt="Customer profile user interface"
                />
                <div className="absolute inset-0 lg:left-auto mix-blend-overlay bg-gradient-to-r from-purple-800 to-indigo-600 lg:w-massive"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default AlternatingFeatureSections;
