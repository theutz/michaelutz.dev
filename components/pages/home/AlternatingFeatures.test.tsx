import { render } from "@testing-library/react"
import { AlternatingFeatureSections } from "./AlternatingFeatures"

describe("<AlternatingFeatureSections />", () => {
  it("renders", () => {
    // Arrange

    // Act
    const { asFragment } = render(<AlternatingFeatureSections />)

    // Assert
    expect(asFragment()).toMatchSnapshot()
  })
})
