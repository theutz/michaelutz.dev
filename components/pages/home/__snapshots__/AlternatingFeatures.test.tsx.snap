// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlternatingFeatureSections /> renders 1`] = `
<DocumentFragment>
  <div
    class="relative pt-16 pb-32 overflow-hidden"
  >
    <div
      aria-hidden="true"
      class="absolute inset-x-0 top-0 h-48 bg-gradient-to-b from-gray-100"
    />
    <div
      class="relative"
    >
      <div
        class="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24"
      >
        <div
          class="max-w-xl px-4 mx-auto sm:px-6 lg:py-16 lg:max-w-none lg:mx-0 lg:px-0"
        >
          <div>
            <div>
              <span
                class="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600"
              >
                <svg
                  aria-hidden="true"
                  class="w-6 h-6 text-white"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 10h4.764a2 2 0 011.789 2.894l-3.5 7A2 2 0 0115.263 21h-4.017c-.163 0-.326-.02-.485-.06L7 20m7-10V5a2 2 0 00-2-2h-.095c-.5 0-.905.405-.905.905 0 .714-.211 1.412-.608 2.006L7 11v9m7-10h-2M7 20H5a2 2 0 01-2-2v-6a2 2 0 012-2h2.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </div>
            <div
              class="mt-6"
            >
              <h2
                class="text-3xl font-extrabold tracking-tight text-gray-900"
              >
                The Humans Have Spoken
              </h2>
              <p
                class="mt-4 text-lg text-gray-500"
              >
                Building tech products is also about building relationships. Here's what some of my co-workers have to say about working with me.
              </p>
              <div
                class="mt-6"
              >
                <button
                  class="flex inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                >
                  Read Another
                </button>
              </div>
            </div>
          </div>
          <div
            class="pt-6 mt-8 border-t border-gray-200"
          >
            <blockquote>
              <div>
                <p
                  class="text-base text-gray-500 whitespace-pre-line"
                >
                  “I worked alongside Michael on the Web Development Team at FRN/UHS. Mike is super friendly and top of the line at his craft. His work in DevOps and JavaScript elevated our team's workflow to the next level, making everyone on our team better at what they do. Mike was our "go-to" guy whenever we had complex development issues. He always communicated clearly and timely while consistently delivering excellent results. It was a privilege to learn from and work with Mike!”
                </p>
              </div>
              <footer
                class="mt-3"
              >
                <div
                  class="flex items-center space-x-3"
                >
                  <div
                    class="relative flex-shrink-0"
                  >
                    <div
                      class="w-6 h-6"
                    >
                      <div
                        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                      >
                        <noscript />
                        <img
                          alt="Benjamin Wright"
                          class="rounded-full"
                          decoding="async"
                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="text-base font-medium text-gray-700"
                  >
                    Benjamin Wright, Director of Ecommerce @ Space Race
                  </div>
                </div>
              </footer>
            </blockquote>
          </div>
        </div>
        <div
          class="mt-12 sm:mt-16 lg:mt-0"
        >
          <div
            class="pl-4 -mr-48 sm:pl-6 md:-mr-16 lg:px-0 lg:m-0 lg:relative lg:h-full"
          >
            <img
              alt="Inbox user interface"
              class="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:left-0 lg:h-full lg:w-auto lg:max-w-none"
              src="https://tailwindui.com/img/component-images/inbox-app-screenshot-1.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-24"
    >
      <div
        class="lg:mx-auto lg:max-w-7xl lg:px-8 lg:grid lg:grid-cols-2 lg:grid-flow-col-dense lg:gap-24"
      >
        <div
          class="max-w-xl px-4 mx-auto sm:px-6 lg:py-32 lg:max-w-none lg:mx-0 lg:px-0 lg:col-start-2"
        >
          <div>
            <div>
              <span
                class="flex items-center justify-center w-12 h-12 rounded-md bg-gradient-to-r from-purple-600 to-indigo-600"
              >
                <svg
                  aria-hidden="true"
                  class="w-6 h-6 text-white"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 3v4M3 5h4M6 17v4m-2-2h4m5-16l2.286 6.857L21 12l-5.714 2.143L13 21l-2.286-6.857L5 12l5.714-2.143L13 3z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </div>
            <div
              class="mt-6"
            >
              <h2
                class="text-3xl font-extrabold tracking-tight text-gray-900"
              >
                Better understand your customers
              </h2>
              <p
                class="mt-4 text-lg text-gray-500"
              >
                Semper curabitur ullamcorper posuere nunc sed. Ornare iaculis bibendum malesuada faucibus lacinia porttitor. Pulvinar laoreet sagittis viverra duis. In venenatis sem arcu pretium pharetra at. Lectus viverra dui tellus ornare pharetra.
              </p>
              <div
                class="mt-6"
              >
                <a
                  class="inline-flex px-4 py-2 text-base font-medium text-white border border-transparent bg-gradient-to-r from-purple-600 to-indigo-600 bg-origin-border rounded-md shadow-sm hover:from-purple-700 hover:to-indigo-700"
                  href="#"
                >
                  Get started
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-12 sm:mt-16 lg:mt-0 lg:col-start-1"
        >
          <div
            class="pr-4 -ml-48 sm:pr-6 md:-ml-16 lg:px-0 lg:m-0 lg:relative lg:h-full"
          >
            <img
              alt="Customer profile user interface"
              class="w-full shadow-xl rounded-xl ring-1 ring-black ring-opacity-5 lg:absolute lg:right-0 lg:h-full lg:w-auto lg:max-w-none"
              src="https://tailwindui.com/img/component-images/inbox-app-screenshot-2.jpg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
